File: PhotonTracker/src/java/servlet/Photon.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package servlet;

import java.util.Date;
import java.util.HashMap;
import java.util.HashSet;

/**
 *
 * @author shengxu
 */
public class Photon {

    private String id;
    private long lastTime;

    public Photon(String id, long lastTime) {
        this.id = id;
        this.lastTime = lastTime;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public long getLastTime() {
        return lastTime;
    }

    public void setLastTime(long lastTime) {
        this.lastTime = lastTime;
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }

    @Override
    public boolean equals(Object x) {
        if (((Photon)x).id.equals(this.id)) {
            return true;
        } else {
            return false;
        }

    }
}

File: PhotonTracker/web/photon.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


function updateTime(response) {

    var lastTime = response.time;
    var time = document.getElementById(response.id.toString());
    var interval = new Date().getTime() - lastTime;
    console.log(interval);
    if (interval < 20000) {
        time.innerHTML = "" + interval / 1000 + "seconds ago";
    }
    else {
        time.innerHTML = "" + interval / 1000 + "seconds ago. \n" + "Suspected Failure";
    }


}

function clickButton(id) {
    var req = newXMLHttpRequest();
    req.onreadystatechange = getReadyStateHandler(req, updateTime);
    req.open("POST", "/PhotonTracker/PhotonServlet", true);
    req.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
    req.send("id=" + id);
}

